metrics/segmentStats.py

```diff
@@ -0,0 +1,506 @@
+
+           SegmentStats.lengthDict - A dictionary with key as the (branching index (nth branch from the start node),
+                                           start node, end node of the branch) value = length of the branch
+
+           SegmentStats.tortuosityDict - A dictionary with key as the (branching index (nth branch from the start node),
+                                             start node, end node of the branch) value = tortuosity of the branch
+
+           SegmentStats.totalSegments - total number of branches (segments between branch and branch points, branch and end points)
+
+           SegmentStats.typeGraphdict - A dictionary with the nth disjoint graph as the key and the type of graph as the value
+
+           SegmentStats.avgBranching - Average branching index (number of branches at a branch point) of the network
+
+           SegmentStats.countEndPoints- nodes with only one other node connected to them
+
+           SegmentStats.countBranchPoints - nodes with more than 2 nodes connected to them
+
+           SegmentStats.contractionDict - A dictionary with key as the (branching index (nth branch from the start node), start node,
+                                                end node of the branch) value = contraction of the branch
+
+           SegmentStats.hausdorffDimensionDict - A dictionary with key as the (branching index (nth branch from the start node), start node,
+                                                       end node of the branch) value = hausdorff dimension of the branch
+
+
+    Notes
+    --------
+
+    tortuosity = curveLength / curveDisplacement
+    contraction = curveDisplacement / curveLength (better becuase there is no change of instability (undefined) in case of cycles)
+    Hausdorff Dimension = np.log(curveLength) / np.log(curveDisplacement) https://en.wikipedia.org/wiki/Hausdorff_dimension
+    Type of subgraphs:
+    0 = if graph a single node
+    1 = if graph is a single cycle
+    2 = line (highest degree in the subgraph is 2)
+    3 = undirected acyclic graph
+    4 = undirected cyclic graph
+
+    """
+    def __init__(self, networkxGraph):
+        self.networkxGraph = networkxGraph
+        # intitialize all the instance variables of SegmentStats class
+        self.contractionDict = {}
+        self.countDict = {}
+        self.cycleInfoDict = {}
+        self.hausdorffDimensionDict = {}
+        self.isolatedEdgeInfoDict = {}
+        self.lengthDict = {}
+        self.tortuosityDict = {}
+        self.typeGraphdict = {}
+
+        self.cycles = 0
+        self.edgesUntraced = 0
+        self.totalSegments = 0
+        self._sortedSegments = []
+        self._visitedSources = []
+        self._visitedPaths = []
+        # list of _disjointGraphs
+        self._disjointGraphs = list(nx.connected_component_subgraphs(self.networkxGraph))
+
+    def _getLengthAndRemoveTracedPath(self, subGraphSkeleton, path, isCycle=0, remove=1):
```

Comment:
`isCycle` and `remove` should have boolean defaults, not int.

Also, why is `remove` optional? What does this function do if we're not removing the traced path?